CafeNet/Data/Mappers/UserMapper.cs

```diff
@@ -18,18 +18,51 @@ public static User ToUser(RegisterUserRequest request)
             };
         }
 
-        public static UserDto ToUserDto(User user)
+        public static User PatchUser(User user, PatchUserRequest request)
         {
-            return new UserDto
-            {
-                Id = user.Id,
-                Name = user.Name,
-                Username = user.Username,
-                Password = user.Password,
-                Role = user.Role.ToString(),
-                LocationId = user.LocationId,
-                LocationAddress = user.Location?.Address ?? "Unassigned"
-            };
+            if (request.Name is not null)
+                user.Name = request.Name;
+
+            if (request.Username is not null)
+                user.Username = request.Username;
+
+            if (request.LocationId is not null)
+                user.LocationId = request.LocationId;
+
+            user.Version = uint.Parse(request.Version);
+
+            return user;
+        }
+
+        public static User PatchUser(User user, PatchOwnProfileRequest request)
```

Comment:
I believe, this mapping method and the one above could be simplified with coalescing operator:
```suggestion
public static User PatchUser(User user, PatchOwnProfileRequest request)
{
    user.Name = request.Name ?? user.Name;
    user.Username = request.Username ?? user.Username;
    user.Password = request.Password != null
        ? BCrypt.Net.BCrypt.EnhancedHashPassword(request.Password, 13)
        : user.Password;
    user.LocationId = request.LocationId ?? user.LocationId;
    user.Version = uint.Parse(request.Version);

    return user;
}
```